google/cloud/firestore_v1/base_document.py

```diff
@@ -503,6 +504,48 @@ def _to_protobuf(self) -> Optional[Document]:
         return _helpers.document_snapshot_to_protobuf(self)
 
 
+class DocumentSnapshotList(list):
+    """A list of DocumentSnapshot.
+
+    This is a subclass of the built-in list. A new property `explain_metrics`
+    is added to return the query profile results.
+
+    Args:
+        reference (:class:`~google.cloud.firestore_v1.document.DocumentReference`):
+            A document reference corresponding to the document that contains
+            the data in this snapshot.
+        docs (list[DocumentSnapshot]):
+            The list of DocumentSnapshot.
+        explain_options
+            (Optional[:class:`~google.cloud.firestore_v1.query_profile.ExplainOptions`]):
+            Options to enable query profiling for this query. When set,
+            explain_metrics will be available on the returned generator.
+        explain_metrics (Optional[query_profile.ExplainMetrics]):
+            Query profile results.
+    """
+
+    def __init__(
+        self,
+        docs: list,
+        explain_options: Optional[query_profile.ExplainOptions] = None,
+        explain_metrics: Optional[query_profile.ExplainMetrics] = None,
+    ):
+        super().__init__(docs)
+        self._explain_options = explain_options
+        self._explain_metrics = explain_metrics
+
+    @property
+    def explain_options(self):
+        return self._explain_options
```

Comment:
Does StreamGenerator also expose `explain_options`?

Reply:
Good catch! Added the property.